PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/AddNewHDInsightReaderClusterForm.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) Microsoft Corporation
+ * 
+ * All rights reserved. 
+ * 
+ * MIT License
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files 
+ * (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, 
+ * publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, 
+ * subject to the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF 
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR 
+ * ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH 
+ * THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+ 
+package com.microsoft.azuretools.azureexplorer.forms;
+
+import java.util.function.Predicate;
+
+import org.eclipse.swt.widgets.Shell;
+
+import com.microsoft.azure.hdinsight.common.ClusterManagerEx;
+import com.microsoft.azure.hdinsight.sdk.cluster.ClusterDetail;
+import com.microsoft.azure.hdinsight.sdk.cluster.HDInsightAdditionalClusterDetail;
+import com.microsoft.azure.hdinsight.sdk.cluster.IClusterDetail;
+import com.microsoft.azure.hdinsight.serverexplore.hdinsightnode.HDInsightRootModule;
+import com.microsoft.azuretools.azurecommons.helpers.NotNull;
+
+public class AddNewHDInsightReaderClusterForm extends AddNewClusterForm {
+	private ClusterDetail selectedClusterDetail;
+	private String defaultStorageRootPath;
+	
+	public AddNewHDInsightReaderClusterForm(Shell parentShell, @NotNull HDInsightRootModule hdinsightRootModule, @NotNull ClusterDetail selectedClusterDetail) {
+		super(parentShell, hdinsightRootModule);
+		this.selectedClusterDetail = selectedClusterDetail;
+		this.defaultStorageRootPath = selectedClusterDetail.getDefaultStorageRootPath();
+	}
+	
+	@Override
+	protected void customizeUI() {
+		this.clusterNameField.setText(selectedClusterDetail.getName());
+		this.clusterNameField.setEditable(false);
+	}
+	
+	private Predicate<IClusterDetail> getSelectedLinkedHdiClusterPredicate(@NotNull String selectedClusterName) {
```

Comment:
I would like suggest to declare a `boolean` return value function, which would be more clear.

Reply:
In the class `MessageDialog`, the dialog return code is the button id, which is defined by the index of button labels array. In our case, the button labels is `new String[] { "Link this cluster", "Cancel" }`, therefore, the `Link the cluster` button id is 0 while the `Cancel` button id is 1.
```
    @Override
 protected void buttonPressed(int buttonId) {
        setReturnCode(buttonId);
        close();
    }
```

---
In reply to: [285934934](https://github.com/microsoft/azure-tools-for-java/pull/3138#discussion_r285934934) [](ancestors = 285934934)